src/Connector.sol

```diff
@@ -42,23 +51,29 @@ interface EscrowLike {
 struct Pool {
     uint64 poolId;
     uint256 createdAt;
+    bool isActive;
+}
+
+struct CFGTranche {
+    uint64 poolId;
+    bytes16 trancheId;
 }
 
-struct Tranche {
-    address token;
-    uint128 latestPrice; // Fixed point integer with 27 decimals
-    uint256 lastPriceUpdate;
-    // TODO: the token name & symbol need to be stored because of the separation between adding and deploying tranches.
-    // This leads to duplicate storage (also in the ERC20 contract), ideally we should refactor this somehow
-    string tokenName;
-    string tokenSymbol;
-    uint8 decimals;
+struct UserTrancheValues {
+    uint256 maxDeposit;
+    uint256 maxMint;
+    uint256 maxWithdraw;
+    uint256 maxRedeem;
+    uint256 openRedeem;
+    uint256 openInvest;
 }
 
-contract CentrifugeConnector {
-    mapping(address => uint256) public wards;
+contract CentrifugeConnector is Auth {
+
     mapping(uint64 => Pool) public pools;
-    mapping(uint64 => mapping(bytes16 => Tranche)) public tranches;
+    mapping(uint64 => mapping(bytes16 => address)) public tranches;
+    mapping(address => CFGTranche) public cfgTranches; // maps evm tranches to CFG tranches
```

Comment:
I think we should design the connector implementation to already support multiple currencies, even if we don't support it on Centrifuge Chain yet. I _think_ the only thing that's needed is changing
```solidity
mapping(uint64 => mapping(bytes16 => address)) public tranches;
```
to
```solidity
mapping(uint64 => mapping(bytes16 => mapping(uint128 => address))) public tranches;
```
basically adding the currency id to the mapping. And then updating the references to this mapping in the `handle*` methods.